### PR TITLE
completed assessment

### PR DESCRIPTION
**Pull request by CathrineC**
# Submitting Week 3 Assessment

### ✅ Completed Files
Mark the following as confirmation that the files in this repository are complete.
- [X] Coding Challenges
- [X] Interview Questions
- [X] React Debugging

### 📝 Feedback
If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -21,9 +21,47 @@ const fibLength1 = 6
...
+    it ("returns an array that length containing the numbers of the Fibonacci sequence", () => {
+        expect(fibSequence(fibLength1)).toEqual([1, 1, 2, 3, 5, 8])
+        expect(fibSequence(fibLength2)).toEqual([1, 1, 2, 3, 5, 8, 13, 21, 34, 55])
+    })
+})
+        // FAIL  ./code-challenges.test.js
+        // fibSequence
+        //   ✕ returns an array that length containing the numbers of the Fibonacci sequence (1 ms)
+        // ● fibSequence › returns an array that length containing the numbers of the Fibonacci sequence
+        //   ReferenceError: fibSequence is not defined
```
Comment (SunkissedQueen) [inline code-challenges.test.js:34]: great structure

File: code-challenges.test.js [hunk 2]
```diff
@@ -21,9 +21,47 @@ const fibLength1 = 6
...
 // b) Create the function that makes the test pass.
 
+// PSEUDO CODE:
+// - create a function that takes in a number greater than 2
+// - output: returns an array with length containing the numbers of the Fibonacci sequence
+// - create a variable with an empty array where the numbers can be placed into
+// - build a for loop that will run the following logic 'num' number of times 
+//      - create a condition that will check if there's atleast 2 elements in the array, if there's none, it will push 1 to the array
+//      - once there are atleast 2 elements, the default logic will get the sum of the previous element and the current element and push the sum into the end of the array
+
```
Comment (SunkissedQueen) [inline code-challenges.test.js:45]: yes to the pseudocode

File: code-challenges.test.js [hunk 3]
```diff
@@ -36,8 +74,45 @@ const fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
...
+      item => typeof item === 'number' && (item % 2 !== 0) 
+    ).sort(function (a, b) {
+        return a - b;
+      }) 
+        return numbersOnly
+    }
+    console.log(onlyOdd(fullArr1))
+    console.log(onlyOdd(fullArr2))
+    console.log("=====OddNumbersOnly=====")
+
```
Comment (SunkissedQueen) [inline code-challenges.test.js:111]: beautiful work

File: code-challenges.test.js [hunk 4]
```diff
@@ -52,5 +127,47 @@ const numbersToAdd2 = [0, 7, -8, 12]
...
+        }else {
+            let totalNum = (numArray[i] + currentSum[i - 1]) 
+                currentSum.push(totalNum)
+        }   
+    } return currentSum
+} 
+console.log(accSum(numbersToAdd1))
+console.log(accSum(numbersToAdd2))
+console.log(accSum(numbersToAdd3))
+
```
Comment (SunkissedQueen) [inline code-challenges.test.js:170]: great work

File: code-challenges.test.js [hunk 5]
```diff
@@ -52,5 +127,47 @@ const numbersToAdd2 = [0, 7, -8, 12]
...
+            return currentSum
+        }else if (currentSum.length < 1){
+            currentSum.push(numArray[i]) 
+        }else {
+            let totalNum = (numArray[i] + currentSum[i - 1]) 
+                currentSum.push(totalNum)
+        }   
+    } return currentSum
+} 
+console.log(accSum(numbersToAdd1))
```
Comment (SunkissedQueen) [inline code-challenges.test.js:167]: Please comment out the console.logs. They are not required for any of the tests or functions to work. The function call takes place in the expect statement.

File: interview-questions.md [hunk 6]
```diff
@@ -9,47 +9,84 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+I have been exposed in using JSX when I learned React. JSX is JavaScript XML. It allows JavaScript code to be placed inside HTML and have HTML be written inside JavaScript file in React. An HTML has to be inside open and closing tags and to write in JavaScript code it needs to be inside the curly braces {}. JSX are placed in the display part of a parent and child of a Javasript file. An example format would be:
+
+const sampleFunc = () => {
+    return (
+        <>                                   <= JSX starts here
+            <h1>{JavaScript code}</h1>       <= JavasScript inside HTML code, inside {}
+        </>                                  <= JSX ends here
+    )
+}
+
```
Comment (SunkissedQueen) [inline interview-questions.md:21]: very nice

File: interview-questions.md [hunk 7]
```diff
@@ -9,47 +9,84 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 Researched answer:
+In reading through resources online,Yarn was developed by Facebook to resolve issues on npm, however, it is used not to replace it as Yarn still uses the modules used from npm registry. This maybe why it is a short term for Yet Another Resource Negotiator. It's purpose is to manage and monitor workloads. I have read that there are atleast 5 steps that yarn does in a matter of seconds. These are submits the application, allocates a container, connects to the node manager, the node manager launches the container, and the conatiner executes the application.
```
Comment (SunkissedQueen) [inline interview-questions.md:33]: great

File: interview-questions.md [hunk 8]
```diff
@@ -9,47 +9,84 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 3. What is an anonymous function in JavaScript?
 
-Your answer:
+Your answer: 
+Anonymous function is a function without a name or an empty syntax that is added into the attribute onClick in JSX to prevent code to run non-stop once onClick it triggered or clicked. I add the anonymous function to tell the attribute that it needs to stop after return something. This prevents stack overflow. This is what you need to add "() => {}" and below is a sample where you need to add it:
+
+``` <buttong onClick={() => testFunc()}><button/>   <= with anonymous function
+``` <buttong onClick={testFunc()}><button/>         <= without anonymous function
+
+Adding the anonymous function doesn't modify the function of the code, it just makes it stop running non-stop
```
Comment (SunkissedQueen) [inline interview-questions.md:43]: bravo

File: interview-questions.md [hunk 9]
```diff
@@ -9,47 +9,84 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 Researched answer:
+In reading through resources online, class is the blueprint of an object. It organizes objects and groups them based on their similarity. Class is a logical entity. It defines the the value, default value, and/or the behavior of an object. to declare a class, it uses the word "class" and the method constructor() that is used to initialize object properties. Class also allows for us to create methods under the constructor().
+
+Object, on the other hand, is an instance of class. It is the physical entity and can be created many times. It has its own properties and type and may contain many values. The values come with keys and is separated by a colon. They're called key-value pair and each pair is called a property. To access the properties, we use the dot.notation and the array like notation. Values inside the propoerty can be manipulated or changed.
```
Comment (SunkissedQueen) [inline interview-questions.md:65]: quality answers

File: interview-questions.md [hunk 10]
```diff
@@ -9,47 +9,84 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 5. What is the difference between state and props in React?
 
 Your answer:
+I used both state and props when I am coding using React. These two works closely together. State is the information or the variable that can be updated. It is best practice to have state variables inside the parent file. Since react consists of multiple components, these components needs a way to connect information from the parent file. This is where props comes in. Props can read-only information and gives access to components. However, information can only be passed in one direction when using props. There is also called Functional props that can pass back information from child component to the parent file. In my experience, I use the App.js as the parent file and the components are called the child. 
 
 Researched answer:
+In reading through resources online, state is managed within a component like variables in a function. It has an initial value and can be changed. Whereas, Props can be passed from the Parent to child component, however, information that is passed cannot be changed.
```
Comment (SunkissedQueen) [inline interview-questions.md:73]: well done

File: react-debugging.html [hunk 11]
```diff
@@ -81,8 +82,14 @@ <h1>Draw a Card</h1>
               )
             })}
           </div>
+          </>
         )
       }
```
Comment (SunkissedQueen) [inline react-debugging.html:87]: thanks for explaining what you corrected

File: interview-questions.md [hunk 12]
```diff
@@ -9,47 +9,84 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 6. STRETCH: Which is the difference between a div and a span?
+Your answer: 
 
-Your answer:
+I know that both div and span are html tags. I usually use the tag <div></div> to enclose my html code which I often put at the start and end of my html code. However, it can also be used in-between other html tags as long as there is an opening div and a closing div tag. Span, on the other hand, can be used for short texts or objects that you want to specify or highlight or be different from the rest of your content. 
 
 Researched answer:
+In reading through resources online, div is a block-level element meaning you can write your whole html code inside div. It can also be used in chunks or be nensted inside another div. Span, on the other hand, is an inline-level element. Unlike div, span encloses small chunks of text that you want to modify and can be inside a div.
```
Comment (SunkissedQueen) [inline interview-questions.md:81]: nice work on the interview questions

File: interview-questions.md [hunk 13]
```diff
@@ -9,47 +9,84 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 6. STRETCH: Which is the difference between a div and a span?
+Your answer: 
 
-Your answer:
+I know that both div and span are html tags. I usually use the tag <div></div> to enclose my html code which I often put at the start and end of my html code. However, it can also be used in-between other html tags as long as there is an opening div and a closing div tag. Span, on the other hand, can be used for short texts or objects that you want to specify or highlight or be different from the rest of your content. 
 
 Researched answer:
+In reading through resources online, div is a block-level element meaning you can write your whole html code inside div. It can also be used in chunks or be nensted inside another div. Span, on the other hand, is an inline-level element. Unlike div, span encloses small chunks of text that you want to modify and can be inside a div.	
 
 ## Looking Ahead: Terms for Next Week
```
Comment (SunkissedQueen) [inline interview-questions.md:83]: now that you have learned javascript I feel that ruby will be easier to approach

**Review by SunkissedQueen — APPROVED** [review]
Wow, Cathrine, you have really turned around your approach to the interview questions. Your answers are engaging, and you truly show you understand the topic. Your answers throughout the assessment are thoughtful and thorough. Keep up your approach with the interview questions. For the code challenges, I would like to see at least two approaches. Use the for loops that you have grasped, then bring in a high order function to replace the for loop. I know you can do this.

**Comment by CathrineC** [comment]
I appreciate your feedback. Everything that you said helps!

**Pull request merged**